metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/ExecutionStatusAggregator.java

```diff
@@ -65,26 +79,89 @@ public Optional<RunExecution> getWorkflowExecutionFromTaskExecutions(TaskExecuti
-                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, Integer::sum));
+                    .collect(groupingBy(Map.Entry::getKey, Collectors.mapping(Entry::getValue, Collectors.toList())));
+
+            if (statusToMetricsByStatus.isEmpty()) {
+                return Optional.empty();
+            }
+
+            ExecutionStatusMetric executionStatusMetric = new ExecutionStatusMetric();
+            statusToMetricsByStatus.forEach((status, metricsForStatus) -> {
+                MetricsByStatus metricsByStatus = getMetricsByStatusFromMetricsByStatusList(metricsForStatus);
+                executionStatusMetric.getCount().put(status, metricsByStatus);
+            });
 
-            return statusCount.isEmpty() ? Optional.empty() : Optional.of(new ExecutionStatusMetric().count(statusCount));
+            // Calculate metrics over all statuses
+            // Calculate from previous ALL MetricsByStatus (aggregate using aggregated data)
+            List<MetricsByStatus> metricsByStatusesToCalculateAllStatus = statusToMetricsByStatus.get(ExecutionStatusEnum.ALL.name());
+            if (metricsByStatusesToCalculateAllStatus == null) {
+                // If there's no ALL key, calculate from other statuses
+                metricsByStatusesToCalculateAllStatus = statusToMetricsByStatus.values().stream().flatMap(Collection::stream).toList();
+            }
+            MetricsByStatus overallMetricsByStatus = getMetricsByStatusFromMetricsByStatusList(metricsByStatusesToCalculateAllStatus);
+            executionStatusMetric.getCount().put(ExecutionStatusEnum.ALL.name(), overallMetricsByStatus);
+            return Optional.of(executionStatusMetric);
         }
         return Optional.empty();
     }
+
+    /**
+     * Aggregate executions into a MetricsByStatus object. Assumes that all executions have the status
+     * @param executions
+     * @return
+     */
+    private MetricsByStatus getMetricsByStatusFromExecutions(List<RunExecution> executions) {
+        MetricsByStatus metricsByStatus = new MetricsByStatus()
+                .executionStatusCount(executions.size());
+
+        // Figure out metrics by status
+        executionTimeAggregator.getAggregatedMetricFromExecutions(executions).ifPresent(metricsByStatus::setExecutionTime);
+        cpuAggregator.getAggregatedMetricFromExecutions(executions).ifPresent(metricsByStatus::setCpu);
+        memoryAggregator.getAggregatedMetricFromExecutions(executions).ifPresent(metricsByStatus::setMemory);
+        costAggregator.getAggregatedMetricFromExecutions(executions).ifPresent(metricsByStatus::setCost);
+        return metricsByStatus;
+    }
+
+    /**
+     * Aggregate a list of MetricsByStatus objects into a MetricsByStatus object. Assumes that all executions have the same status
+     * @param metricsByStatuses
+     * @return
+     */
+    private MetricsByStatus getMetricsByStatusFromMetricsByStatusList(List<MetricsByStatus> metricsByStatuses) {
```

Comment:
Would it make sense to come up with shorter names for this method (and others like it)?  This method appears to be aggregating, so for example, `aggregateMetricsByStatusList`.  If yes, something that'd be good to do post-release...